### ISSUE TITLE
L107.an_FeedIO_R_C_Sys_Fd_Y fails old new test 

### ISSUE THREAD

**Issue by kdorheim** [JGCRI/gcamdata#929]
### Problem: 

`L107.an_FeedIO_R_C_Sys_Fd_Y ` created by `module_aglu_LA107.an_IMAGE_R_C_Sys_Fd_Y` fails the gcam 5 old new sum test. 

The differences between the gcam5 and dsr output is nontrivial so we cannot do a self test. 

``` 
 gcam5_value         dsr_value        absolute_diff    
 Min.   :  0.7771   Min.   :  0.7771   Min.   :  0.000  
 1st Qu.:  4.8235   1st Qu.:  5.2120   1st Qu.:  0.000  
 Median : 28.6287   Median : 31.3318   Median :  0.000  
 Mean   : 46.8387   Mean   : 48.9180   Mean   :  2.535  
 3rd Qu.:100.0000   3rd Qu.:100.0000   3rd Qu.:  0.000  
 Max.   :514.0544   Max.   :514.0544   Max.   :414.054  
```
![rplot](https://user-images.githubusercontent.com/27299759/38679607-afbc6896-3e53-11e8-9d4d-dbdcfce42e35.png)

### Why: 

A closer look a the output reveals that dsr product has `12480` observations where the value is equal to 100 whereas the gcam 5 data system output only has `10822` observations equal to 100. 

The extra values that are equal to 100 come from the following pipeline (starts at line 169).

```
    L107.an_Feed_Mt_R_C_Sys_Fd_Y %>%
      rename(feedVal = value) %>%
      # add in the corresponding animal production amount
      left_join_error_no_match(L107.an_Prod_Mt_R_C_Sys_Fd_Y,
                               by = c("GCAM_region_ID", "GCAM_commodity", "year", "system", "feed")) %>%
      rename(prodVal = value) %>%
      # calculate the region, commodity, system, feed type, year IO coefficient as feed consumption/animal production
      # note we're dividing by *tiny* numbers, not robust, so round everything to allow old-new comparison
      mutate(value = round(round(feedVal, 10) / round(prodVal, 10), 6)) %>%
      select(-feedVal, -prodVal) %>%
      # Replace NAs with a default value. This is a conservative default IO coefficient
      # for regions without the necessary production data from which to compute one.
      # Tends to be pastoral production in regions with zero pastoral production. If we
      # were to allow this tech in the future (currently it is zero-shareweighted out),
      # we'd need to have something plausible.
      replace_na(list(value = 100)) ->
      # store in a table specifying IO coefficients by region, commodity, system, feed type, and year:
      L107.an_FeedIO_R_C_Sys_Fd_Y
```

This chunk divides a number by  tiny numbers so `round` is used to prevent R from accidently rounding the tiny number to 0 before dividing by them, creating NAs in the data frame. Latter on in the pipeline these NAs are replaced with a default value of 100. 

So the extra 100 value observations come from the dsr chunk creating more NAs during the division step. 

### Ideas / Questions: 

* Replace `round` with some other function, probably `signif`. I have tried this already and it corrects some but not all of the observations. There is also a question of how many significant digits can be used. 
* Considering how unstable this chunk do we need to consider if the gcam 5 data system is correct? Do we need to add an old behavior block? 
* In the mean time do we `temp-data-inject` the gcam 5 output to continue with the xml testing?

**Comment by kdorheim** [comment]
@bpbond, @kvcalvin, and @abigailsnyder any thoughts, ideas, and/or advice would be appreciated, thanks!

**Comment by abigailsnyder** [comment]
uhhhhh I would guess `signif` would be the first place to start; if it doesn't fix the issue, then that's sort of this larger question around precision issues that's come up again and again and needs some sort of more unified fix.

Honestly, the precision issues is probably something that needs to be dealt with in a unified way regardless of whether `signif` fixes this particular issue or not

**Comment by abigailsnyder** [comment]
@bpbond @kvcalvin 
1. charge code for me to deal with _this_ chunk
2. Just to summarize the discussion Kate and I just had in writing, at some point the more "holistic" fix to this precision issue is to go back through and say "We don't have information at the kg level. We don't have information below the ton(nes??) level, so let's round to the nearest ton instead of trying to create information from unrealistic precision."  We're thinking that's v2 and getting this stuff fixed as it comes up within `OLD_DATA_SYSTEM_BEHAVIOR` blocks is the v1/1.1 fix. 
3. I'm pretty sure the answer is "no" but do we want to introduce a new flag beyond `OLD_DATA_SYSTEM_BEHAVIOR` for problems like this? I'm just thinking that this chunk will not have an `else` with an improved solution. So whenever old data system blocks are deleted, this chunk will just fail.

**Comment by bpbond** [comment]
1. Please check with Kali.
2. I completely agree. I want to band-aid, `OLD_DATA_SYSTEM_BEHAVIOR`, whatever us to 1.1, but then this issue needs to be dealt with comprehensively.
3. Hmm...I'm open to it, if useful.

Another option here, as @kdorheim notes above, is just to declare this new behavior "correct" (it means that more values are being set to default 100 versus something else is all). And then later deal with precision issues.

Thanks, all.

**Comment by abigailsnyder** [comment]
I included this in the new comments in the code, and in the PR https://github.com/JGCRI/gcamdata/pull/930

But the reason for the original rounding was to make SheepGoat Pastoral FeedCrops behave in GCAM regions 10 and 29. There was no rounding in the OG data system.  So there's some kind of rounding that needs to happen upstream of L107.an_FeedIO being created. Since that sounds like our V2 fix (ie we don't work with values less than a ton), I just bandaided things now.
